src/spox/_function.py

```diff
@@ -89,11 +92,16 @@ def infer_output_types(self) -> dict[str, Type]:
         }
 
     @property
-    def opset_req(self):
+    def opset_req(self) -> set[tuple[str, int]]:
         node_opset_req = Node.opset_req.fget(self)  # type: ignore
         return node_opset_req | self.func_graph._get_build_result().opset_req
 
-    def update_metadata(self, opset_req, initializers, functions):
+    def update_metadata(
+        self,
+        opset_req: set[tuple[str, int]],
+        initializers: dict[Var, np.ndarray],
+        functions: list["Function"],
```

Comment:
Does this work without the quotation marks  with
```python
from __future__ import annotations
```
?

This applies throughout and may trigger some [ruff reformatting](https://docs.astral.sh/ruff/rules/future-rewritable-type-annotation/).

Reply:
I am left under the impression that the `spox` codebase does not use `__future__` annotations. Do you suggest adding it in all relevant files? Should we leave it be for now and fix it in a different PR? 

Examples: [1](https://github.com/Quantco/spox/blob/main/src/spox/_build.py#L131), [2](https://github.com/Quantco/spox/blob/main/src/spox/_graph.py#L142), [3](https://github.com/Quantco/spox/blob/main/src/spox/_shape.py#L36)

Reply:
We don't use it because we didn't know about it then (or at least I didn't). We should use `__future__`, though, since it makes the type hints so much nicer to write. This PR seems like a good place to do so. Could you add it to the relevant files, please?

Reply:
I knew about `annotations`, but I remember something breaking. 
I think it may have been the places where type annotations are parsed (just operator definitions?).
You can try if it'll work fine now!

Reply:
Yes the part in question that breaks is this [one](https://github.com/Quantco/spox/pull/194/files#diff-7aa15f9b066bdff1950605ed91fda2b81a1802f16cabc3ff64c2649764dc82eaR61-R72), where we parse the type hints. I think this fix ought to work.

Reply:
Follow up: https://github.com/Quantco/spox/pull/194/files#r1869451563